tpkg/util.go

```diff
@@ -734,6 +796,17 @@ func RandBlock(blockBody iotago.BlockBody, api iotago.API, rmc iotago.Mana) *iot
 	}
 }
 
+func RandBasicBlockWithPayload(api iotago.API, payload iotago.ApplicationPayload) *iotago.BasicBlockBody {
```

Comment:
Maybe we should call this one (and friends) `RandBasicBlockBody...` because they don't return an actual Basic Block (which would be a `iotago.Block` with a `BasicBlockBody`).